mypy/semanal.py

```diff
@@ -413,14 +417,17 @@ def visit_func_def(self, defn: FuncDef) -> None:
                 else:
                     # A coroutine defined as `async def foo(...) -> T: ...`
                     # has external return type `Awaitable[T]`.
-                    defn.type = defn.type.copy_modified(
-                        ret_type = self.named_type_or_none('typing.Awaitable',
-                                                           [defn.type.ret_type]))
+                    ret_type = self.named_type_or_none('typing.Awaitable', [defn.type.ret_type])
+                    if ret_type is None:
+                        # We are running tests.
+                        ret_type = self.named_type('typing_full.Awaitable', [defn.type.ret_type])
```

Comment:
Without this line many tests fail (see the comment just above). I think this is because those tests use ``fixtures/typing-full.py``. A better fix would be to always name it ``typing`` in tests, but I think this should be a separate PR.

Reply:
So the issue is that `typing-full.py` is imported as module named `typing_full` in tests? If this is true, this should be fixed (in another PR).

Reply:
Hm, my initial analysis is wrong, the problem is how ``named_type`` works. I will make a PR now.

Reply:
OK, opened https://github.com/python/mypy/pull/4158